tests/test_client.py

```diff
@@ -397,7 +397,7 @@ def test_path_is_not_double_encoded(self):
         self.assertEqual(req.path, "/get/test%20case")
 
         req = ClientRequest('get', "http://0.0.0.0/get/test%2fcase")
-        self.assertEqual(req.path, "/get/test%2fcase")
+        self.assertEqual(req.path, "/get/test/case")
```

Comment:
According to RFC 3986: 
> percent-encoded octets in the ranges of ALPHA
   (%41-%5A and %61-%7A), DIGIT (%30-%39), hyphen (%2D), period (%2E),
   underscore (%5F), or tilde (%7E) should not be created by URI
   producers and, when found in a URI, should be decoded to their
   corresponding unreserved characters by URI normalizers.

So assuming we want to use normalized URIs `%2f` should be converted to `/` as well as other mentioned substitutions.